src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -769,7 +769,7 @@ public Encoding GetEncoding(ILocalRepositoryModel repository, string relativePat
                 }
             }
 
-            return Encoding.Default;
+            return null;
```

Comment:
I believe can we can simply replace this with the following and avoid having to spacial case `null`.

```suggestion
            return new UTF8Encoding(false);
```

This makes the `GetEncoding` responsible for returning `Encoding.UTF8` if the target file has a preamble or `new UTF8Encoding(false)` if the target file doesn't exist or have a preamble. UTF8 without preamble is Git's default text encoding.

It's possible we could always use `new UTF8Encoding(false)` and avoid this issue completely. The question is what a diff looks like between UTF8 files with and without preamble. I didn't know `new UTF8Encoding(false)` was a thing until today.

Reply:
Re: wondering if we could always use `new UTF8Encoding(false)`. Alas, this doesn't work. I tried the following:

```
File.WriteAllText("foo.txt", "foobar", new UTF8Encoding(true));
File.WriteAllText("bar.txt", "foobar", new UTF8Encoding(false));
```

When I diff them together, I see:
![image](https://user-images.githubusercontent.com/11719160/47437219-fed5da80-d79f-11e8-83f6-18b09f497dfc.png)

Reply:
This should change